### PR TITLE
made email field mandatory in proof creation dialog.

### PR DESCRIPTION
**Pull request by damisul**
I've removed old code marked as legacy.
Also did some code cleanup (see comments below).

fixed bug in antispam check (proof created even if check failed)

### CODE REVIEW

File: app/controllers/proof_controller.rb [hunk 1]
```diff
@@ -5,28 +5,25 @@ class ProofController < ApplicationController
...
-        @p.save!
-      end
+    @errors = []
+    unless params[:ziburit] =~ /ביאליק/
+      @errors << I18n.t('proof.create.ziburit_failed')
     end
-    respond_to do |fmt|
-      fmt.html { }
-      fmt.js { flash[:notice] = I18n.t(:proof_thanks_html) }
+    if params[:from].blank?
```
Comment (damisul) [inline app/controllers/proof_controller.rb:12]: Actually as far as I see Proof form says that email is optional, so perhaps we need to update wording there too.

File: app/views/proof/show.html.haml [hunk 2]
```diff
@@ -28,7 +28,7 @@
     .graybox#highlight{style:'font-size: 14pt; border:1px;background-color:#dedede'}
       != sanitize(@p.highlight.gsub("\n","<br/>"))
   %br
-  %h3= t(:proof)+':'
+  %h3= t(:proof_heading)+':'
```
Comment (damisul) [inline app/views/proof/show.html.haml:31]: I've renamed 'proof' line in i18n file, as it conflicted with controller's name

File: config/locales/he.yml [hunk 3]
```diff
@@ -444,7 +444,6 @@ he:
   activate_bit: הוסף תפקיד
   deactivate_bit: בטל תפקיד
   editor_bits: תפקידי עורך
-  proof_thanks_title: קיבלנו, תודה!
```
Comment (damisul) [inline config/locales/he.yml:447]: was used in proof html form I've removed

File: config/locales/he.yml [hunk 4]
```diff
@@ -455,7 +454,6 @@ he:
   proofstatus_wontfix: לא שגוי
   proofstatus_escalated: טעון בדיקה נוספת
   proofstatus_assigned: מוקצה לטיפול
-  proofs: הגהות
```
Comment (damisul) [inline config/locales/he.yml:458]: was not used

File: app/controllers/proof_controller.rb (hunk 1 above)
Comment (abartov) [reply app/controllers/proof_controller.rb:12]: True! I am fixing the string and will commit after merging this.

**Review by abartov — APPROVED** [review]

**Pull request merged**